drivers/modem/gsm_ppp.c

```diff
@@ -606,7 +605,7 @@ static void gsm_finalize_connection(struct gsm_modem *gsm)
 	}
 
 	/* If attach check failed, we should not redo every setup step */
-	if (gsm->attach_retries) {
+	if (gsm->retries) {
```

Comment:
Need to reset the `retries` somewhere so that this part of code can run normally when the modem is restarted.